trio/_tests/tools/test_gen_exports.py

```diff
@@ -121,17 +122,14 @@ def test_process(tmp_path, imports):
 
 @skip_lints
 def test_lint_failure(tmp_path) -> None:
-    """Test that processing properly fails if black or isort does."""
+    """Test that processing properly fails if black or ruff does."""
     try:
         import black  # noqa: F401
-    # there's no dedicated CI run that has astor+isort, but lacks black.
+        import ruff  # noqa: F401
     except ImportError as error:  # pragma: no cover
         skip_if_optional_else_raise(error)
 
     file = File(tmp_path / "module.py", "module")
 
     with pytest.raises(SystemExit):
         run_linters(file, "class not valid code ><")
-
-    with pytest.raises(SystemExit):
-        run_linters(file, "# isort: skip_file")
```

Comment:
If I'm interpreting the comment correctly, this test is no longer checking whether `ruff` fails (the "or" -> as long as `black` fails we don't care whether `ruff` does).

Reply:
Yep that was the intention with these two tests. The first fails from bad syntax and never gets to `isort`, while the latter errors because all files were skipped. (Isort is quite permissive, I couldn’t find something that just syntax errors.)

Reply:
Split up the tests as of 6cc59b7